aspnetcore/signalr/get-started-signalr-core/sample/wwwroot/js/chat.js

```diff
@@ -0,0 +1,26 @@
+﻿const connection = new signalR.HubConnection('/chathub');
+
+connection.on('Send', (timestamp, user, message) => {
+    const encodedUser = user;
+    const encodedMsg = message;
+    const listItem = document.createElement('li');
+    listItem.innerHTML = timestamp + ' <b>' + encodedUser + '</b>: ' + encodedMsg;
+    document.getElementById('messages').appendChild(listItem);
+});
+
+document.getElementById('send').addEventListener('click', event => {
+    const msg = document.getElementById('message').value;
+    const usr = document.getElementById('user').value;
+
+    connection.invoke('Send', usr, msg).catch(err => showErr(err));
+    event.preventDefault();
+});
+
+function showErr(msg) {
+    const listItem = document.createElement('li');
+    listItem.setAttribute("style", "color: red");
+    listItem.innerHTML = msg;
```

Comment:
Hrm, let's use `innerText` here. `innerHTML` could technically be used to inject HTML. Technically the error content comes from SignalR, which should be trustworthy, but it's better to use `innerText` because it should be treated as plain text. Also, we should use `.toString`.

So this would be

```javascript
listItem.innerText = msg.toString();
```